packages/rum-core/test/common/apm-server.spec.js

```diff
@@ -123,62 +128,23 @@ describe('ApmServer', function() {
     configService.setConfig({
       serviceName: 'serviceName'
     })
-
-    spyOn(apmServer, '_postJson')
     apmServer.init()
+    spyOn(apmServer, '_postJson')
 
-    let trs = generateTransaction(19)
+    const trs = generateTransaction(19)
     trs.forEach(apmServer.addTransaction.bind(apmServer))
-    expect(apmServer.transactionQueue.items.length).toBe(19)
+    expect(apmServer.queue.items.length).toBe(19)
     expect(apmServer._postJson).not.toHaveBeenCalled()
 
-    apmServer.transactionQueue.flush()
+    apmServer.queue.flush()
     expect(apmServer._postJson).toHaveBeenCalled()
-    expect(apmServer.transactionQueue.items.length).toBe(0)
-  })
-
-  it('should init queue if not initialized before', function(done) {
```

Comment:
Don't we still need this test for the event queue?

Reply:
This was there because we were using it for testing purposes. right now its under the assumption that `apmServer.init` is called always when we are depending on the queue in tests.

Reply:
I think we should still test how the APM server behaves if we (or the user doesn't call the init) function. IMO, apm server class should handle these cases as well.

Reply:
Not sure why this is important, Please explain the scenario when this would happen. APM server class would still be unusable without init.

Reply:
My only concern is that if the user doesn't call the init, the apm server is not going to throw any exception, and instead gracefully handles.

We don't have to add these exact tests back btw, we just need to make sure apm server doesn't throw errors if it's not initialised.

Reply:
> My only concern is that if the user doesn't call the init, the apm server is not going to throw any exception, and instead gracefully handles.

I am not sure if users would need to use our APMServer class alone, but if we need a graceful handling, We can do two things

1. We can do a `noop` for throttleEvents 

(OR)

2. Initialize queue during ApmServer class construction itself and get rid of init method and makes testing also a lot easier. 

WYDT?

Reply:
I went with option 1 as it requires less changes and added tests covering the case you mentioned.